### DISCUSSION TITLE
一個好的程式該有甚麼特點

### DISCUSSION THREAD

**Discussion by Tomlord1122** [1101-GS4719-C/Python-Community-GS4719-C#209] [General]
最近看了很多寫程式的文章，想統整一下一個好的程式的判定標準
大概分為兩項:空間複雜度與時間複雜度
空間複雜度是在說開發這個程式，花了多大的空間，佔用了多少記憶體，而通常又會把空間分為固定成本+變動成本來計算。
變動成本像是程式執行會呼叫另外一個函式或是在跑loop時，額外產生的空間。
固定成本就是程式碼的大小了

時間複雜度=開發這個程式的時間+執行這個程式所花的時間
通常教科書都只會討論執行程式所花的時間，(因為討論開發她的時間好像沒甚麼意義，很憑個人的實力)
而計算確切的時間成本也挺大的，所以又以執行多少個"指令"當作評斷時間複雜度的依據。

**Comment by wax75397** [comment]
我覺得時間複雜度怪怪的，如果兩個人花費不同時間開發出幾乎一樣的程式，他們的程式卻不一樣好嗎?

**Reply by Tomlord1122** [reply]
所以大家最後都以執行時間來判斷了，但教科書還是有寫開發時間

**Reply by xCk27x** [reply]
兩個功能幾乎相同的程式除了運作時間的快慢以外，在未來維護時的難易度也能作為好壞的判斷標準。

**Comment by luo910227** [comment]
我覺得一個好程式就算空間複雜度與時間複雜度都很低，但最基本的還是必須要能夠做好debug，並且能使使用者使用起來是友善的。

**Comment by jcpan821** [comment]
我覺得是可讀性，能不能淺顯易懂是關鍵

**Comment by etin2468** [comment]
我也覺得可讀性重要，因為很多大專案的程式碼交接就是因為前面的人寫的可讀性很差，所以後面的人很痛苦...

**Comment by Lynn12378** [comment]
確實可讀性超重要，有時候小組專題要整合的時候就會看不太懂其他人的Code，導致要花費很多時間磨合

**Comment by Jimmy911218** [comment]
謝謝分享但如果能做出成果我還是優先，殼毒性我想放在第二哈哈

**Comment by Hchejxnc** [comment]
空間複雜度以及時間複雜度兩個確實很重要，但是我覺得能不能讓其他人讀懂更重要，之前看到別人的Code隨便定義一個變數，卻沒有解釋它，花了一段時間才知道那個變數代表什麼..
